var/spack/repos/builtin/packages/python/package.py

```diff
@@ -286,8 +286,8 @@ class Python(Package):
     patch("tkinter-3.10.patch", when="@3.10.0:3.10 ~tkinter")
     patch("tkinter-3.11.patch", when="@3.11.0:3.11 ~tkinter")
 
-    # Ensure that distutils chooses correct compiler option for RPATH on cray:
-    patch("cray-rpath-3.1.patch", when="@3 platform=cray")
+    # Ensure that distutils chooses correct compiler option for RPATH:
+    patch("rpath-non-gcc.patch")
```

Comment:
```suggestion
    patch("rpath-non-gcc.patch", when="@:3.11")
```
The distutils module will be removed in Python 3.12 so there won't be anything to patch.